pkg/apis/planetscale/v2/vitessbackupschedule_types.go

```diff
@@ -0,0 +1,206 @@
+	// by the given amount of seconds.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty"`
+
+	// ConcurrencyPolicy specifies ho to treat concurrent executions of a Job.
+	// Valid values are:
+	// - "Allow" (default): allows CronJobs to run concurrently;
+	// - "Forbid": forbids concurrent runs, skipping next run if previous run hasn't finished yet;
+	// - "Replace": cancels currently running job and replaces it with a new one.
+	// +optional
+	// +kubebuilder:example="Forbid"
+	ConcurrencyPolicy ConcurrencyPolicy `json:"concurrencyPolicy,omitempty"`
+
+	// AllowedMissedRun defines how many missed run of the schedule will be allowed before giving up on finding the last job.
+	// If the operator's clocked is skewed and we end-up missing a certain number of jobs, finding the last
+	// job might be very consuming, depending on the frequency of the schedule and the duration during which
+	// the operator's clock was misbehaving, also depending on how laggy the clock is, we can end-up with thousands
+	// or missed runs. For this reason, AllowedMissedRun, which is set to 100 by default, will allow us to give up finding
+	// the last job and simply wait for the next job on the schedule.
+	// Unless you are experiencing issue with missed runs due to a misconfiguration of the clock, we recommend leaving
+	// this field to its default value.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	AllowedMissedRun *int `json:"allowedMissedRun,omitempty"`
+
+	// JobTimeoutMinute defines after how many minutes a job that has not yet finished should be stopped and removed.
+	// Default value is 10 minutes.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=10
+	JobTimeoutMinute int32 `json:"jobTimeoutMinute,omitempty"`
+}
+
+// VitessBackupScheduleStrategy defines how we are going to take a backup.
+// The VitessBackupSchedule controller uses this data to build the vtctldclient
+// command line that will be executed in the Job's pod.
+type VitessBackupScheduleStrategy struct {
+	// Name of the backup strategy.
+	Name BackupStrategyName `json:"name"`
+
+	// KeyspaceShard defines the keyspace and shard on which we want to take a backup.
+	// It has to be formatted as follows: <keyspace>/<shard>
+	// This field is mandatory if we have picked the strategy BackupShard.
+	// This field will be ignored if we have picked the strategy BackupTablet.
+	// +optional
+	// +kubebuilder:example="commerce/-"
+	KeyspaceShard string `json:"keyspaceShard,omitempty"`
+
+	// TabletAlias is the tablet alias we want to take a backup on.
+	// This field is mandatory if we have picked the strategy BackupTablet.
+	// This field will be ignored if we have picked the strategy BackupShard.
+	// +optional
+	// +kubebuilder:example="zone1-0000000102"
+	TabletAlias string `json:"tabletAlias"`
```

Comment:
i kinda question the value of cron-scheduled `Backup`:

 * it's not uncommon to delete a PVC/Pod because it's in bad shape, and having to update backup schedules anytime that happens will be annoying.
 * a image/config rollout may switch a tablet from replica to primary, which means the next time a backup schedule runs it will take the primary offline.

would say it's not worth adding this strategy with these two potential footguns, unless it is a hotly requested feature.

Reply:
That's a good point. I think it is safe to drop this strategy especially if I add what you are mentioning in https://github.com/planetscale/vitess-operator/pull/553#discussion_r1607039861. Having two strategies: `BackupShard` and `BackupCluster`, would give enough flexibility for most use cases. I am thinking we can even add a `BackupKeyspace` strategy, which is a good in-between solution between `BackupShard` and `BackupCluster`.

Reply:
See comment https://github.com/planetscale/vitess-operator/pull/553#issuecomment-2130317283, which explains the new strategies.